src/planner/builder/project.go

```diff
@@ -432,3 +442,24 @@ func GetProject(root PlanNode) string {
 	}
 	return project
 }
+
+// replaceCol replace the field based on colMap.
+// such as: select tmp+1 from (select a+1 as tmp,b from t1)t;
+// `tmp+1` will be overwritten as 'a+1+1 as `tmp+1`'.
+func replaceSelect(field selectTuple, colMap map[string]selectTuple) (selectTuple, error) {
+	var err error
+	if len(field.info.referTables) > 0 {
+		field.info, err = replaceCol(field.info, colMap)
+		if err != nil {
+			return field, err
+		}
+		if expr, ok := field.expr.(*sqlparser.AliasedExpr); ok {
+			expr.Expr = field.info.expr
```

Comment:
expr.Expr have changed, 
 is `expr.Expr = field.info.expr` verbose? 
or in the structure of the `selectTuple`
expr, info.expr, just leave one?

Reply:
`expr` and `info.expr` are different.  `expr`'type is `sqlparser.SelectExpr`, `info.expr` is `sqlparser.Expr`. see different in `ast.go` plz.